lib/parse/block-element.js

```diff
@@ -13,20 +13,29 @@ const TEXT_ELEMENTS = {
   blockquote: 'blockquote'
 };
 
+const isPullQuote = (elm) => {
+  return elm.classList.contains('q');
+};
+
 // inject parse to avoid recursive requires
 export default (parse, text) => {
   return (elm, textOpts) => {
     const tagName = elm.tagName.toLowerCase();
 
     if (BLOCK_ELEMENTS.has(tagName)) {
+      const type = TEXT_ELEMENTS[tagName] || 'block';
       const blockElement = {
-        type: TEXT_ELEMENTS[tagName] || 'block',
+        type,
         children: []
       };
       if (elm.childNodes.length) {
         parse(elm.childNodes, text(textOpts, elm), blockElement.children);
       }
 
+      if (type === 'blockquote') {
+        blockElement.pullQuote = isPullQuote(elm);
```

Comment:
we don't do this anyway - change existing objects - instead we're create new objects (treating them as they're immutable and make sure they're getting optimized by V8).

In this case we could do something like this
```js

const createBlockElement = () => {
  if (type === 'blockquote') {
    return {
      type,
      children: [],
      pullQuote: isPullQuote(elm)
    }
  }

    return {
      type,
      children: [],
      pullQuote: isPullQuote(elm)
    };
}

const blockElemnt = createBlockElement();
```